packages/eas-json/src/accessor.ts

```diff
@@ -38,16 +38,26 @@ const customErrorMessageHandlers: ((err: ValidationError) => void)[] = [
 ];
 
 export class EasJsonAccessor {
-  private easJsonPath: string;
+  private easJsonPath: string | undefined;
 
   private isJson5 = false;
   private easJson: EasJson | undefined;
   private easJsonRawContents: string | undefined;
   private easJsonRawObject: any | undefined;
   private easJsonPatched: boolean = false;
 
-  constructor(projectDir: string) {
-    this.easJsonPath = EasJsonAccessor.formatEasJsonPath(projectDir);
+  private constructor() {}
+
+  public static fromProjectPath(projectDir: string): EasJsonAccessor {
+    const accessor = new EasJsonAccessor();
+    accessor.easJsonPath = EasJsonAccessor.formatEasJsonPath(projectDir);
+    return accessor;
+  }
+
+  public static fromRawString(rawEasJson: string): EasJsonAccessor {
+    const accessor = new EasJsonAccessor();
+    accessor.easJsonRawContents = rawEasJson;
+    return accessor;
   }
```

Comment:
Consider refactoring to use constructor overload overloading. This is cleaner in my opinion.
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBGAhjAFnAZlCI4HImq4DcAUCQMYA2iAzjXAKK0BSNEAdgILnkCmd0OAG8SASDBRgAN2S84vFm3YAFZCgBccGjEnsA5nAA+cAK7sAJr3TB2vc6XGSZMOQpqsOAJUQB3AMIcLuwwNJraugbGZpbWtvZkjtKycOQc4SbkMNAAFEIIWABWvJkAIsBQcAC+mnkSEEWl5WE6NgaVAJQOEkkuKWk6GVlQufKKXr4BwbzB9NXCo+5K3v6B0yHNEVWdYt3Ocqns6Zk5tYXFMGVQADQLHuzLk0EhVTX59eeXAPwbrUS3SxNVjNvloWvotsIxKJUMAaAA6Nx3VSoOAAXjeDQu5TgADIcYwxlwePw2FA4ehoCBkExFhxkShsnVMZdtqJoShYQjCQ8gc90YiASspjMHJUEmATAAjSjAcig5CyjCU6mE+mMs6NKA-fTtbUGERwQ1wURQXgwExQdgINRwgoQGzq96am64Ny2pS4baG0RisQAej9cFQchAZpQEHM9ElvEoEB8cEQprgEDAMGAHEQlDgwHQcAAnhATHBzBxcPByIm5IhJYX4CYaK0gxz6DLJXAfMBULW4AAVADK-sDkpM8E7cFNAEcTOV+HAqQBrRvBvqHAbHCoS6WKxB6RA2HZSmVy7QKuWYbDKDUwNVMj5NUERXUE2lEvgCCoiNmm82WuC2eM0nc3BvqSIy3pqWyiuKh6KieaZnlgIDLH2YJ6NkArjEKTyhA+rRPoBSjASSgifiaZoWla-7PkBxLviMGH3ICwrPB0UGVEAA